acceptance/tests/wan-federation/wan_federation_test.go

```diff
@@ -190,11 +209,259 @@ func TestWANFederation(t *testing.T) {
+			}
+
+			primaryContext := env.DefaultContext(t)
+			secondaryContext := env.Context(t, 1)
+
+			primaryHelmValues := map[string]string{
+				"global.datacenter": primaryDatacenter,
+
+				"global.tls.enabled":   "true",
+				"global.tls.httpsOnly": strconv.FormatBool(c.secure),
+
+				"global.federation.enabled":                "true",
+				"global.federation.createFederationSecret": "true",
+
+				"global.acls.manageSystemACLs":       strconv.FormatBool(c.secure),
+				"global.acls.createReplicationToken": strconv.FormatBool(c.secure),
+
+				"connectInject.enabled":  "true",
+				"connectInject.replicas": "1",
+
+				"meshGateway.enabled":  "true",
+				"meshGateway.replicas": "1",
+
+				"global.enableConsulNamespaces":               "true",
+				"connectInject.consulNamespaces.mirroringK8S": "true",
+			}
+
+			if cfg.UseKind {
+				primaryHelmValues["meshGateway.service.type"] = "NodePort"
+				primaryHelmValues["meshGateway.service.nodePort"] = "30000"
+			}
+
+			releaseName := helpers.RandomName()
+
+			// Install the primary consul cluster in the default kubernetes context
+			primaryConsulCluster := consul.NewHelmCluster(t, primaryHelmValues, primaryContext, cfg, releaseName)
+			primaryConsulCluster.Create(t)
+
+			// Get the federation secret from the primary cluster and apply it to secondary cluster
+			federationSecretName := fmt.Sprintf("%s-consul-federation", releaseName)
+			logger.Logf(t, "Retrieving federation secret %s from the primary cluster and applying to the secondary", federationSecretName)
+			federationSecret, err := primaryContext.KubernetesClient(t).CoreV1().Secrets(primaryContext.KubectlOptions(t).Namespace).Get(context.Background(), federationSecretName, metav1.GetOptions{})
+			require.NoError(t, err)
+			federationSecret.ResourceVersion = ""
+			federationSecret.Namespace = secondaryContext.KubectlOptions(t).Namespace
+			_, err = secondaryContext.KubernetesClient(t).CoreV1().Secrets(secondaryContext.KubectlOptions(t).Namespace).Create(context.Background(), federationSecret, metav1.CreateOptions{})
+			require.NoError(t, err)
+
+			var k8sAuthMethodHost string
+			// When running on kind, the kube API address in kubeconfig will have a localhost address
+			// which will not work from inside the container. That's why we need to use the endpoints address instead
+			// which will point the node IP.
+			if cfg.UseKind {
+				// The Kubernetes AuthMethod host is read from the endpoints for the Kubernetes service.
+				kubernetesEndpoint, err := secondaryContext.KubernetesClient(t).CoreV1().Endpoints("default").Get(context.Background(), "kubernetes", metav1.GetOptions{})
+				require.NoError(t, err)
+				k8sAuthMethodHost = fmt.Sprintf("%s:%d", kubernetesEndpoint.Subsets[0].Addresses[0].IP, kubernetesEndpoint.Subsets[0].Ports[0].Port)
+			} else {
+				k8sAuthMethodHost = k8s.KubernetesAPIServerHostFromOptions(t, secondaryContext.KubectlOptions(t))
+			}
```

Comment:
It looks like this code has been copied in a few places but there's a nearly identical helper (adds `https://` to host) [here](https://github.com/hashicorp/consul-k8s/blob/4d40591d4eb93b07f0eb7770623cff457a082890/acceptance/framework/k8s/helpers.go#L88). Could we use that instead?

Reply:
I'll try it out, not sure if `https://` might have a negative effect on the test

Reply:
looks like it works 🚀 !